src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -1063,7 +1065,8 @@ public void TrimExcess(int capacity)
 
         ICollection IDictionary.Values => (ICollection)Values;
 
-        object? IDictionary.this[object key] // TODO-NULLABLE: https://github.com/dotnet/csharplang/issues/2384
+        [DisallowNull]
```

Comment:
Why do we have to put it here rather than in the setter? Just for readability?

Reply:
Allowing the attributes on getters and setters would require adding an attribute usage Method target, which then means these attributes could be used in a bunch of places where they don't make sense.  It also helps with consistency, providing one place to put these.

Reply:
Ah, yeah. I forgot we discussed this on the review for the attributes, too much things discussed in the past days 😄